src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointIndexer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xypoint;
+
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYPoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import org.apache.lucene.document.XYPointField;
+import org.apache.lucene.geo.XYPoint;
+import org.apache.lucene.index.IndexableField;
+import org.opensearch.geometry.Point;
+import org.opensearch.index.mapper.AbstractGeometryFieldMapper;
+import org.opensearch.index.mapper.ParseContext;
+
+/**
+ * Converts points into Lucene-compatible form for indexing in a xy_point field.
+ */
+@AllArgsConstructor
+public class XYPointIndexer implements AbstractGeometryFieldMapper.Indexer<List<ParsedXYPoint>, List<? extends XYPoint>> {
+    private final String fieldName;
+
+    /**
+     * @param points list of parsed xypoints
+     * @return list of xypoints that are converted from opensearch to lucene type
+     */
+    @Override
+    public List<? extends XYPoint> prepareForIndexing(@NonNull List<ParsedXYPoint> points) {
+        Objects.requireNonNull(points, "XYPoint cannot be null");
```

Comment:
This is redundant i guess since you already added @NonNull annotation.

Reply:
I guess this is the difference in the error msg with and without adding 
```
Objects.requireNonNull(points, "XYPoint cannot be null");
```
Without - java.lang.NullPointerException: points is marked non-null but is null

With - java.lang.NullPointerException: XYPoint cannot be null

Reply:
Yes, you can keep one null check instead of two.

Reply:
Yeah, I will remove the annotation.